packages/pigeon/lib/dart_generator.dart

```diff
@@ -252,15 +275,29 @@ void _writeFlutterApi(
                 indent.writeln('// ignore message');
                 call = 'api.${func.name}()';
               } else {
-                final String argType =
-                    _addGenericTypes(func.arguments[0].type, nullTag);
                 indent.writeln(
-                  'assert(message != null, \'Argument for $channelName was null. Expected $argType.\');',
+                  'assert(message != null, \'Argument for $channelName was null.\');',
                 );
+                const String argsArray = 'args';
                 indent.writeln(
-                  'final $argType input = (message as $argType$nullTag)$unwrapOperator;',
-                );
-                call = 'api.${func.name}(input)';
+                    'final List<Object$nullTag> $argsArray = (message as List<Object$nullTag>$nullTag)$unwrapOperator;');
+                final Iterable<String> argNames = indexMap(
+                    func.arguments,
+                    (int count, NamedType arg) =>
+                        arg.name.isEmpty ? 'arg$count' : 'arg_${arg.name}');
```

Comment:
Unless I'm missing something, this lambda is a direct duplicate of `_getSafeArgumentName`.

Reply:
Done.